x-pack/plugins/remote_clusters/server/routes/api/remote_clusters/register_add_route.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { licensePreRoutingFactory } from '../../../lib/license_pre_routing_factory';
+import { isEsErrorFactory } from '../../../lib/is_es_error_factory';
+import { callWithRequestFactory } from '../../../lib/call_with_request_factory';
+import { wrapEsError, wrapUnknownError } from '../../../lib/error_wrappers';
+
+export function registerAddRoute(server) {
+  const isEsError = isEsErrorFactory(server);
+  const licensePreRouting = licensePreRoutingFactory(server);
+
+  server.route({
+    path: '/api/remote_clusters',
+    method: 'POST',
+    handler: async (request) => {
+      const callWithRequest = callWithRequestFactory(server, request);
+      const { name, ...rest } = request.payload;
+
+      const addClusterPayload = {
+        persistent: {
+          cluster: {
+            remote: {
+              [name]: {
+                ...rest,
+              }
+            }
+          }
+        }
+      };
+
+      try {
+        return await callWithRequest('cluster.putSettings', { body: addClusterPayload });
+      } catch (err) {
+        if (isEsError(err)) {
+          throw wrapEsError(err);
+        }
+
+        throw wrapUnknownError(err);
```

Comment:
I'm seeing errors both thrown and returned from various endpoints in other apps. @joshdover Does it matter if we use `throw` or `return` here?

Reply:
I just tested adding clusters with invalid names and am getting the correct ES error back, so looks like `throw` works, but am open to changing it if Josh says `return` is a better standard 🙂